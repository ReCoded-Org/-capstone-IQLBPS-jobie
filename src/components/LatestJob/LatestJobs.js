import { useTranslation } from "react-i18next";
import { useState } from "react";
import CustomButton from "../UI/Button/CustomButton";

const LatestJobs = ({ headers, data }) => {
  const { t } = useTranslation();
  const [fullTime, setFullTime] = useState(false);
  const [partTime, setPartTime] = useState(false);
  const [remote, setRemote] = useState(false);
  const handleFullTime = () => {
    setFullTime((prev) => !prev);
  };
  const handlePartTime = () => {
    setPartTime((prev) => !prev);
  };
  const handleRemote = () => {
    setRemote((prev) => !prev);
  };
  const filteredJobs = data.filter(
    (job) =>
      (partTime && job.partTime && !fullTime && !remote) ||
      (fullTime && job.fullTime && !partTime && !remote) ||
      (remote && job.remote && !partTime && !fullTime) ||
      (!remote && partTime && job.partTime && fullTime && job.fullTime) ||
      (remote && job.remote && partTime && job.partTime && !fullTime) ||
      (remote && job.remote && !partTime && fullTime && job.fullTime) ||
      !(remote || partTime || fullTime)
  );
  return (
    <div className="bg-gray-100">
      <div className="max-w-[70%] large:max-w-[70%] medium:max-w-[80%] small:max-w-[85%] h-full mx-auto py-10 small:py-5">
        <h1 className="text-3xl small:text-xl text-secondary mb-10 small:mb-5 font-semibold">
          {t('latest-jobs')}
        </h1>
        <div className="flex justify-end">
          <label htmlFor="full-time" className="mr-4 small:mr-2 small:text-sm">
            <input
              type="checkbox"
              id="full-time"
              className="mr-2 small:mr-1"
              onChange={handleFullTime}
              checked={fullTime}
            />
            {t('full-time')}
          </label>
          <label htmlFor="part-time" className="mr-4 small:mr-2 small:text-sm">
            <input
              type="checkbox"
              id="part-time"
              className="mr-2 small:mr-1"
              onChange={handlePartTime}
              checked={partTime}
            />
            {t('part-time')}
          </label>
          <label htmlFor="remote" className="small:text-sm">
            <input
              type="checkbox"
              id="remote"
              className="mr-2 small:mr-1"
              onChange={handleRemote}
              checked={remote}
            />
            {t('remote')}
          </label>
        </div>
        <div className="overflow-x-auto shadow-md">
          <table className="w-full text-sm text-left text-gray-500 dark:text-gray-400">
            <thead className="text-xs text-secondary bg-gray-50 dark:bg-gray-700 dark:text-gray-400">
              <tr>
                {headers.map((header) => {
                  if (header.title === "") {
                    return <td key={header.id} />;
                  }
                  return (
                    <th
                      scope="col"
                      className="py-3 px-6 small:py-2 small:px-3"
                      key={header.id}
                    >
                      {header.title}
                    </th>
                  );
                })}
              </tr>
            </thead>
            <tbody>
              {filteredJobs.map((rowData) => {
                return (
                  <tr
                    className="bg-white dark:bg-gray-800 small:text-xs"
                    key={rowData.id}
                  >
                    <th
                      scope="row"
                      className="py-4 px-6 small:py-1 small:px-2 font-medium text-gray-900 whitespace-nowrap dark:text-white"
                    >
                      {rowData.position}
                    </th>
                    <td className="py-4 px-6 small:py-1 small:px-2">
                      {rowData.salary}
                    </td>
                    <td className="py-4 px-6 small:py-1 small:px-2">
                      {rowData.date}
                    </td>
                    <td className="py-4 px-6 small:py-1 small:px-2 w-20">
                      <CustomButton btnName={t('apply-job')} />
                    </td>
                  </tr>
                );
              })}
            </tbody>
          </table>
        </div>
      </div>
    </div>
  );
};
export default LatestJobs;
