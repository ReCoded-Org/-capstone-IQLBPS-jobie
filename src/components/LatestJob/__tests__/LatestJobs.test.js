import React from "react";
import renderer from "react-test-renderer";
import LatestJobs from "../LatestJobs";
const data = [
  {
    position: "Web Developer f",
    salary: "2000$-3000$",
    date: "10/10/2022",
    fullTime: true,
    partTime: false,
    remote: false,
    id: 1,
  },
  {
    position: "Web Developer p",
    salary: "2000$-3000$",
    date: "10/10/2022",
    fullTime: false,
    partTime: true,
    remote: false,
    id: 2,
  },
  {
    position: "Web Developer fr",
    salary: "2000$-3000$",
    date: "10/10/2022",
    fullTime: true,
    partTime: false,
    remote: true,
    id: 3,
  },
  {
    position: "Web Developer pr",
    salary: "2000$-3000$",
    date: "10/10/2022",
    fullTime: false,
    partTime: true,
    remote: true,
    id: 4,
  },
  {
    position: "Web Developer fr",
    salary: "2000$-3000$",
    date: "10/10/2022",
    fullTime: true,
    partTime: false,
    remote: true,
    id: 5,
  },
  {
    position: "Web Developer r",
    salary: "2000$-3000$",
    date: "10/10/2022",
    fullTime: false,
    partTime: false,
    remote: true,
    id: 6,
  },
  {
    position: "Web Developer fp",
    salary: "2000$-3000$",
    date: "10/10/2022",
    fullTime: true,
    partTime: true,
    remote: false,
    id: 7,
  },
];
const header = [
  { title: "Position", id: 1 },
  { title: "Salary", id: 2 },
  { title: "Date", id: 3 },
  { title: "", id: 4 },
];
it("renders Latest Jobs correctly", () => {
  const tree = renderer
    .create(<LatestJobs headers={header} data={data} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
