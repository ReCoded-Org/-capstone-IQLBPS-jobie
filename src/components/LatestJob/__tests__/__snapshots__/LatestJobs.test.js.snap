// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Latest Jobs correctly 1`] = `
<div
  className="bg-gray-100"
>
  <div
    className="max-w-[70%] large:max-w-[70%] medium:max-w-[80%] small:max-w-[85%] h-full mx-auto py-10 small:py-5"
  >
    <h1
      className="text-3xl small:text-xl text-secondary mb-10 small:mb-5 font-semibold"
    >
      Latest Jobs
    </h1>
    <div
      className="flex justify-end"
    >
      <label
        className="mr-4 small:mr-2 small:text-sm"
        htmlFor="full-time"
      >
        <input
          checked={false}
          className="mr-2 small:mr-1"
          id="full-time"
          onChange={[Function]}
          type="checkbox"
        />
        Full Time
      </label>
      <label
        className="mr-4 small:mr-2 small:text-sm"
        htmlFor="part-time"
      >
        <input
          checked={false}
          className="mr-2 small:mr-1"
          id="part-time"
          onChange={[Function]}
          type="checkbox"
        />
        Part Time
      </label>
      <label
        className="small:text-sm"
        htmlFor="remote"
      >
        <input
          checked={false}
          className="mr-2 small:mr-1"
          id="remote"
          onChange={[Function]}
          type="checkbox"
        />
        Remote
      </label>
    </div>
    <div
      className="overflow-x-auto shadow-md"
    >
      <table
        className="w-full text-sm text-left text-gray-500 dark:text-gray-400"
      >
        <thead
          className="text-xs text-secondary bg-gray-50 dark:bg-gray-700 dark:text-gray-400"
        >
          <tr>
            <th
              className="py-3 px-6 small:py-2 small:px-3"
              scope="col"
            >
              Position
            </th>
            <th
              className="py-3 px-6 small:py-2 small:px-3"
              scope="col"
            >
              Salary
            </th>
            <th
              className="py-3 px-6 small:py-2 small:px-3"
              scope="col"
            >
              Date
            </th>
            <td />
          </tr>
        </thead>
        <tbody>
          <tr
            className="bg-white dark:bg-gray-800 small:text-xs"
          >
            <th
              className="py-4 px-6 small:py-1 small:px-2 font-medium text-gray-900 whitespace-nowrap dark:text-white"
              scope="row"
            >
              Web Developer f
            </th>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              2000$-3000$
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              10/10/2022
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2 w-20"
            >
              <button
                className="w-25 h-7 bg-white text-accent border-2 
 border-accent rounded-3xl px-5 small:w-15 small:h-5 
 small:rounded-xs small:px-2 hover:border-primary hover:text-primary"
                type="button"
              >
                Apply
              </button>
            </td>
          </tr>
          <tr
            className="bg-white dark:bg-gray-800 small:text-xs"
          >
            <th
              className="py-4 px-6 small:py-1 small:px-2 font-medium text-gray-900 whitespace-nowrap dark:text-white"
              scope="row"
            >
              Web Developer p
            </th>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              2000$-3000$
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              10/10/2022
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2 w-20"
            >
              <button
                className="w-25 h-7 bg-white text-accent border-2 
 border-accent rounded-3xl px-5 small:w-15 small:h-5 
 small:rounded-xs small:px-2 hover:border-primary hover:text-primary"
                type="button"
              >
                Apply
              </button>
            </td>
          </tr>
          <tr
            className="bg-white dark:bg-gray-800 small:text-xs"
          >
            <th
              className="py-4 px-6 small:py-1 small:px-2 font-medium text-gray-900 whitespace-nowrap dark:text-white"
              scope="row"
            >
              Web Developer fr
            </th>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              2000$-3000$
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              10/10/2022
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2 w-20"
            >
              <button
                className="w-25 h-7 bg-white text-accent border-2 
 border-accent rounded-3xl px-5 small:w-15 small:h-5 
 small:rounded-xs small:px-2 hover:border-primary hover:text-primary"
                type="button"
              >
                Apply
              </button>
            </td>
          </tr>
          <tr
            className="bg-white dark:bg-gray-800 small:text-xs"
          >
            <th
              className="py-4 px-6 small:py-1 small:px-2 font-medium text-gray-900 whitespace-nowrap dark:text-white"
              scope="row"
            >
              Web Developer pr
            </th>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              2000$-3000$
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              10/10/2022
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2 w-20"
            >
              <button
                className="w-25 h-7 bg-white text-accent border-2 
 border-accent rounded-3xl px-5 small:w-15 small:h-5 
 small:rounded-xs small:px-2 hover:border-primary hover:text-primary"
                type="button"
              >
                Apply
              </button>
            </td>
          </tr>
          <tr
            className="bg-white dark:bg-gray-800 small:text-xs"
          >
            <th
              className="py-4 px-6 small:py-1 small:px-2 font-medium text-gray-900 whitespace-nowrap dark:text-white"
              scope="row"
            >
              Web Developer fr
            </th>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              2000$-3000$
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              10/10/2022
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2 w-20"
            >
              <button
                className="w-25 h-7 bg-white text-accent border-2 
 border-accent rounded-3xl px-5 small:w-15 small:h-5 
 small:rounded-xs small:px-2 hover:border-primary hover:text-primary"
                type="button"
              >
                Apply
              </button>
            </td>
          </tr>
          <tr
            className="bg-white dark:bg-gray-800 small:text-xs"
          >
            <th
              className="py-4 px-6 small:py-1 small:px-2 font-medium text-gray-900 whitespace-nowrap dark:text-white"
              scope="row"
            >
              Web Developer r
            </th>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              2000$-3000$
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              10/10/2022
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2 w-20"
            >
              <button
                className="w-25 h-7 bg-white text-accent border-2 
 border-accent rounded-3xl px-5 small:w-15 small:h-5 
 small:rounded-xs small:px-2 hover:border-primary hover:text-primary"
                type="button"
              >
                Apply
              </button>
            </td>
          </tr>
          <tr
            className="bg-white dark:bg-gray-800 small:text-xs"
          >
            <th
              className="py-4 px-6 small:py-1 small:px-2 font-medium text-gray-900 whitespace-nowrap dark:text-white"
              scope="row"
            >
              Web Developer fp
            </th>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              2000$-3000$
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2"
            >
              10/10/2022
            </td>
            <td
              className="py-4 px-6 small:py-1 small:px-2 w-20"
            >
              <button
                className="w-25 h-7 bg-white text-accent border-2 
 border-accent rounded-3xl px-5 small:w-15 small:h-5 
 small:rounded-xs small:px-2 hover:border-primary hover:text-primary"
                type="button"
              >
                Apply
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
